docs/source/best-practices/checkpointing.rst

```diff
@@ -74,8 +86,53 @@ To submit this array job to the scheduler, use the following command::
 
    sbatch submit_mdrun_array.bash
 
-Python TensorFlow checkpointing example
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Checkpointing using DMTCP
+--------------------------
+
+`DMTCP <https://dmtcp.sourceforge.io/>`_ (Distributed MultiThreaded checkpointing) is a tool that lets you checkpoint without the need to change your code. It Works with most Linux applications such as Python, Matlab, R, GUI, MPI, etc.
+The program runs in the background of your program without significant performance loss and saves the process states into checkpoint files. DMTCP is available on the cluster ::
+
+ module avail dmtcp
+ module show dmtcp
+ module load dmtcp/2.6.0
+
+As DMTCP runs in the background, it requires some changes to your Shell script. For examples of how to checkpoint with DMTCP visit `here <https://github.com/northeastern-rc/training-checkpointing/tree/main/Exercise_3>`_.
```

Comment:
Because DMTCP runs in the background, changes to your shell script are required.